homeassistant/components/wsdot/sensor.py

```diff
@@ -144,12 +144,15 @@ def extra_state_attributes(self) -> dict[str, Any] | None:
         return None
 
 
-def _parse_wsdot_timestamp(timestamp):
+def _parse_wsdot_timestamp(timestamp: Any) -> datetime | None:
```

Comment:
```suggestion
def _parse_wsdot_timestamp(timestamp: str) -> datetime | None:
```
Wouldn't this work? 

If it doesn't then this is okay as it isn't great but it at least gets us to a better way and to improve from here.

No need to keep bugging about this

Reply:
No, because at this point the data pulled out of raw json could still be None and is still typed as Any. This function should only ever receive `str | None`, but the only way to do that and make mypy happy is to narrow the json Any before calling. However, that still leaves the None case, so it seemed like a lot of work to still leave unexpected cases for datetime parsing and I just went ahead and made this function handle any possible input that is not a valid datetime string as None.